aesm-client/src/imp/windows.rs

```diff
@@ -0,0 +1,393 @@
+            this: *mut InnerAesmInterface,
+            ps_cap: *mut u64,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    sgx_register: Option<
+        unsafe extern "system" fn(
+            this: *mut InnerAesmInterface,
+            white_list_cert: *mut u8,
+            white_list_cert_size: u32,
+            registration_data_type: u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    proxy_setting_assist: Option<
+        unsafe extern "system" fn(
+            this: *mut InnerAesmInterface,
+            proxy_info: *mut u8,
+            proxy_size: u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    query_sgx_status: Option<
+        unsafe extern "system" fn(
+            this: *mut InnerAesmInterface,
+            sgx_status: *mut u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    get_whitelist_size: Option<
+        unsafe extern "system" fn(
+            this: *mut InnerAesmInterface,
+            white_list_size: *mut u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    get_white_list: Option<
+        unsafe extern "system" fn(
+            this: *mut InnerAesmInterface,
+            white_list: *mut u8,
+            buf_size: u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    get_sec_domain_id: Option<
+        unsafe extern "system" fn(
+            this: *mut InnerAesmInterface,
+            sec_domain_id: *mut u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    switch_sec_domain: Option<
+        unsafe extern "system" fn(
+            this: *mut InnerAesmInterface,
+            sec_domain_id: u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    get_epid_provision_status: Option<
+        unsafe extern "system" fn(
+            this: *mut InnerAesmInterface,
+            epid_pr_status: *mut u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    get_platform_service_status: Option<
+        unsafe extern "system" fn(
+            this: *mut InnerAesmInterface,
+            pse_status: *mut u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+}
+
+const CLSID_AESMINTERFACE: CLSID = CLSID {
+    Data1: 0x82367CAB,
+    Data2: 0xF2B9,
+    Data3: 0x461A,
+    Data4: [0xB6, 0xC6, 0x88, 0x9D, 0x13, 0xEF, 0xC6, 0xCA],
+};
+const IID_IAESMINTERFACE: IID = IID {
+    Data1: 0x50AFD900,
+    Data2: 0xF309,
+    Data3: 0x4557,
+    Data4: [0x8F, 0xCB, 0x10, 0xCF, 0xAB, 0x80, 0x2C, 0xDD],
+};
+
+trait HresultExt: Sized {
+    fn into_io_error(self) -> IoResult<Self>;
+}
+
+impl HresultExt for HRESULT {
+    fn into_io_error(self) -> IoResult<HRESULT> {
+        if SUCCEEDED(self) {
+            Ok(self)
+        } else {
+            Err(IoError::from_raw_os_error(self))
+        }
+    }
+}
+
+#[derive(Clone, Debug, Default)]
+pub struct AesmClient {
+    dummy_private_var: u32,
```

Comment:
Use: `_private: ()`

1. I don't know why there is no warning about this field being unused, but it could be because of the `derive`s. Prefixing a field with `_` silences that warning.
2. `()` doesn't take up any space in memory.